imap/jmap_mail_submission.c

```diff
@@ -555,6 +555,35 @@ static void _emailsubmission_create(jmap_req_t *req,
         envelope = NULL;
     }
 
+    json_t *onSend = json_object_get(emailsubmission, "onSend");
+    if (JNOTNULL(onSend)) {
+        const char *field;
+        json_t *jval;
+
+        jmap_parser_push(&parser, "onSend");
+        json_object_foreach(onSend, field, jval) {
+            if (!strcmp(field, "moveToMailboxId")) {
+                if (JNOTNULL(jval) && !json_is_string(jval)) {
+                    jmap_parser_invalid(&parser, "moveToMailboxId");
+                }
+            }
+            else if (!strcmp(field, "setKeywords")) {
+                const char *keyword;
+                json_t *jbool;
+
+                jmap_parser_push(&parser, "setKeywords");
+                json_object_foreach(jval, keyword, jbool) {
+                    if (!json_is_boolean(jbool) ||
+                        !jmap_email_keyword_is_valid(keyword)) {
+                        jmap_parser_invalid(&parser, keyword);
+                    }
+                }
```

Comment:
isn't an

    else jmap_parser_invalid(parser, field)

missing here for unknown arguments?

Reply:
Yep